### PR TITLE
grey out roomsettings that you can't edit

### PR DESCRIPTION
**Pull request by ara4n**

### CODE REVIEW

**Comment by ara4n** [comment]
hopefully fixes:
https://github.com/vector-im/vector-web/issues/1088
https://github.com/vector-im/vector-web/issues/740
https://github.com/vector-im/vector-web/issues/764
and possibly others...

File: src/components/views/rooms/RoomSettings.js [hunk 1]
```diff
@@ -321,18 +321,18 @@ module.exports = React.createClass({
 
         var state_default = (parseInt(power_levels ? power_levels.state_default : 0) || 0);
 
-        var room_aliases_level = state_default;
-        if (events_levels['m.room.aliases'] !== undefined) {
-            room_aliases_level = events_levels['m.room.aliases'];
```
Comment (dbkr) [inline src/components/views/rooms/RoomSettings.js]: I moved this out to the js sdk in https://github.com/matrix-org/matrix-react-sdk/pull/187 / https://github.com/matrix-org/matrix-js-sdk/pull/94 so maybe we should just merge that and use it.

**Comment by dbkr** [comment]
otherwise looks good

**Comment by ara4n** [comment]
i've done the switch, but it's caused a regression because roomState.maySendStateEvent() doesn't check to see if the user is a guest (in which case they certainly can't send state events).  Suggestions welcome on the right fix for this.

**Pull request merged**